agent/ebs/watcher.go

```diff
@@ -106,12 +123,31 @@ func (w *EBSWatcher) HandleEBSResourceAttachment(ebs *apiebs.ResourceAttachment)
 	volumeId := ebs.GetAttachmentProperties(apiebs.VolumeIdKey)
 	ebsAttachment, ok := w.agentState.GetEBSByVolumeId(volumeId)
 	if ok {
-		log.Infof("EBS Volume attachment already exists. Skip handling EBS attachment %v.", ebs.EBSToString())
+		log.Debugf("EBS Volume attachment already exists. Skip handling EBS attachment %v.", ebs.EBSToString())
 		return ebsAttachment.StartTimer(func() {
 			w.handleEBSAckTimeout(volumeId)
 		})
 	}
 
+	// start EBS CSI Driver Managed Daemon
+	if runningCsiTask := w.taskEngine.GetDaemonTask(md.EbsCsiDriver); runningCsiTask != nil {
+		log.Debugf("engine ebs CSI driver is running with taskID: %v", runningCsiTask.GetID())
+	} else {
+		if ebsCsiDaemonManager, ok := w.taskEngine.GetDaemonManagers()[md.EbsCsiDriver]; ok {
+			if csiTask, err := ebsCsiDaemonManager.CreateDaemonTask(); err != nil {
+				// fail attachment and return
+				log.Errorf("Unable to start ebsCsiDaemon for task %s in the engine: error: %s", csiTask.GetID(), err)
```

Comment:
`CreateDaemonTask` can return `nil` task on error and calling `csiTask.GetID()` will panic in that case.